### PR TITLE
feat: (CXSPA-9820) - Add access continuum test for Applied Promotions

### PR DESCRIPTION
**Pull request by petarmarkov9449**
Closes: [CXSPA-9820](https://jira.tools.sap/browse/CXSPA-9820)

Add access continuum test for Applied Promotions

### CODE REVIEW

File: projects/storefrontapp-e2e-cypress/cypress/e2e/accessibility/applied-promotions.a11y-e2e.cy.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    cy.get('cx-add-to-cart')
+      .findByText(/Add To Cart/i)
+      .click();
+
+    cy.get('cx-added-to-cart-dialog').should('be.visible');
+    cy.get('cx-added-to-cart-dialog button.close').click({ force: true });
+
+    cy.get('cx-mini-cart').click();
+
+    cy.get('.cx-promotions').a11yRunContinuumTest();
```
Comment (Pio-Bar) [inline projects/storefrontapp-e2e-cypress/cypress/e2e/accessibility/applied-promotions.a11y-e2e.cy.ts]: I see the intention is to run this on the cart page however, with this setup the test runs on PDP instead.

**Review by Pio-Bar — CHANGES_REQUESTED** [review]

File: projects/storefrontapp-e2e-cypress/cypress/e2e/accessibility/applied-promotions.a11y-e2e.cy.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    cy.get('cx-added-to-cart-dialog .cx-promotions').a11yRunContinuumTest();
+
+    cy.get('cx-added-to-cart-dialog button.close').click({ force: true });
+  });
+
+  it('should show promotions on Cart page', () => {
+    cy.get('cx-mini-cart').click();
+
+    cy.get('.cart-details-wrapper').should('exist');
```
Comment (Pio-Bar) [inline projects/storefrontapp-e2e-cypress/cypress/e2e/accessibility/applied-promotions.a11y-e2e.cy.ts]:
```suggestion
    cy.get('.cart-details-wrapper')
```

File: projects/storefrontapp-e2e-cypress/cypress/e2e/accessibility/applied-promotions.a11y-e2e.cy.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    cy.get('.cx-promotions').a11yRunContinuumTest();
```
Comment (Pio-Bar) [inline projects/storefrontapp-e2e-cypress/cypress/e2e/accessibility/applied-promotions.a11y-e2e.cy.ts]: Selectors before the tests return more than one element. With our current setup Continuum will only scan the first one. Do we want to scan each or is this enough? 🤔

**Review by Pio-Bar — CHANGES_REQUESTED** [review]

File: projects/storefrontapp-e2e-cypress/cypress/e2e/accessibility/applied-promotions.a11y-e2e.cy.ts (hunk 3 above)
Comment (petarmarkov9449) [reply projects/storefrontapp-e2e-cypress/cypress/e2e/accessibility/applied-promotions.a11y-e2e.cy.ts]: There is only one `cx-promotions` in `cart` page, isn't there? 🤔

Comment (Pio-Bar) [reply projects/storefrontapp-e2e-cypress/cypress/e2e/accessibility/applied-promotions.a11y-e2e.cy.ts]: There's two. Same goes for the previous one. In the 'add to cart' test only an empty div gets scanned.

image

Comment (petarmarkov9449) [reply projects/storefrontapp-e2e-cypress/cypress/e2e/accessibility/applied-promotions.a11y-e2e.cy.ts]: Updated the test for all promotions here.

**Review by Pio-Bar — APPROVED** [review]

**Review by Zeyber — APPROVED** [review]
